### PR TITLE
make build image and version configurable via env

### PR DESCRIPTION
**Pull request by OskarStark**
Not sure if this [commit](https://github.com/continuouspipe/continuouspipe/pull/64/commits/12e3976283a1213cd8a49ad521acea040cb2a780) is needed. I made it to stay BC, but it would be easier to move the timeout param to the last position in the signature.

### CODE REVIEW

File: .env.dist [hunk 1]
```diff
@@ -38,13 +38,19 @@ FIREBASE_SERVICE_ACCOUNT_PATH=/runtime/keys/firebase.json
...
+#+only-if(GOOGLE_CONTAINER_BUILDER_VERSION):(GOOGLE_CONTAINER_BUILDER_ENABLED=true)
 #+file-to-propagate(GOOGLE_CONTAINER_BUILDER_SERVICE_ACCOUNT_PATH)
 #
 BUILDER_DEFAULT_ENGINE=docker
 
 GOOGLE_CONTAINER_BUILDER_ENABLED=false
 GOOGLE_CONTAINER_BUILDER_SOURCE_BUCKET=
 GOOGLE_CONTAINER_BUILDER_PROJECT_ID=
+GOOGLE_CONTAINER_BUILDER_BUILD_TIMEOUT=3600
+GOOGLE_CONTAINER_BUILDER_IMAGE=gcr.io/continuous-pipe-1042/cloud-builder
```
Comment (sroze) [inline .env.dist]: Can you put the Docker Hub one here?

File: api/src/ContinuousPipe/Builder/GoogleContainerBuilder/HttpBuildCreator.php [hunk 2]
```diff
@@ -29,16 +29,23 @@ class HttpBuildCreator implements BuildCreator
...
+     */
+    private $builderImage;
+
     public function __construct(
         GuzzleHttpClientFactory $httpClientFactory,
         string $googleProjectId,
         string $googleSourceArtifactBucket,
-        int $maximumAllowedBuildTime = 3600
+        int $maximumAllowedBuildTime = 3600,
+        string $builderImage = 'gcr.io/continuous-pipe-1042/cloud-builder'
```
Comment (sroze) [inline api/src/ContinuousPipe/Builder/GoogleContainerBuilder/HttpBuildCreator.php]: Same, let's use the Docker Hub one here.

File: api/src/ContinuousPipe/Builder/GoogleContainerBuilder/HttpGoogleContainerBuildClient.php [hunk 3]
```diff
@@ -45,20 +45,28 @@ class HttpGoogleContainerBuildClient implements GoogleContainerBuilderClient
...
     public function __construct(
         ArchiveBuilder $archiveBuilder,
         ArtifactManager $artifactManager,
         GuzzleHttpClientFactory $httpClientFactory,
         ManifestFactory $manifestFactory,
         BuildCreator $buildCreator,
-        string $googleProjectId = null
+        string $googleProjectId = null,
+        string $builderVersion = 'v8'
+
```
Comment (sroze) [inline api/src/ContinuousPipe/Builder/GoogleContainerBuilder/HttpGoogleContainerBuildClient.php]: Can you remove this space?

File: .env.dist (hunk 1 above)
Comment (OskarStark) [reply .env.dist]: I can, no need to stay BC with the old default value?

File: api/src/ContinuousPipe/Builder/GoogleContainerBuilder/HttpBuildCreator.php (hunk 2 above)
Comment (OskarStark) [reply api/src/ContinuousPipe/Builder/GoogleContainerBuilder/HttpBuildCreator.php]: Ok

What do you think about putting allowed build time to the and and break BC?

File: .env.dist (hunk 1 above)
Comment (sroze) [reply .env.dist]: No, it isn't working anyway as it was private 😓

File: api/src/ContinuousPipe/Builder/GoogleContainerBuilder/HttpBuildCreator.php (hunk 2 above)
Comment (sroze) [reply api/src/ContinuousPipe/Builder/GoogleContainerBuilder/HttpBuildCreator.php]: > What do you think about putting allowed build time to the and and break BC?

I don't get the question :)

Comment (sroze) [reply api/src/ContinuousPipe/Builder/GoogleContainerBuilder/HttpBuildCreator.php]: Well, as long as you change the service definition, that should be OK. I don't know anyone having such a deep customisation yet so I believe we can go ahead.

File: api/src/ContinuousPipe/Builder/GoogleContainerBuilder/HttpGoogleContainerBuildClient.php (hunk 3 above)
Comment (OskarStark) [reply api/src/ContinuousPipe/Builder/GoogleContainerBuilder/HttpGoogleContainerBuildClient.php]: done

**Comment by OskarStark** [comment]
LGTM @sroze 👍

**Comment by OskarStark** [comment]
Does the api image builds automatically

**Comment by sroze** [comment]
> Does the api image builds automatically

yep, it does.

**Pull request merged**